File: _Job-Search/InterviewPractice-master/InterviewPractice-master/Python/selectionSort.py
def selctionSort(list):
    for i in range(len(list)):
        minVal = min(list[i:])
        minIndex = list[i:].index(minVal)

        list[i+minIndex] = list[i]
        list[i] = minVal
    print(list)

def main():
    myList = [4, 2, 1, 10, 5, 3, 100, 69]
    nullList = []
    singleList = [3]
    selctionSort(myList)

main()

File: _Job-Search/InterviewPractice-master/InterviewPractice-master/Java/src/selectionSort.java
/**
 * Created by ahadsheriff on 3/2/17.
 */
public class selectionSort {

    public static int[] selectionSort(int[] list) {
        for (int i = 0; i < list.length-1; i++) {
            int index = i;
            for (int j = i + 1; j < list.length; j++) {
                if(list[j] < list[index]) {
                    index = j;
                }
            }
            int min = list[index];
            list[index] = list[i];
            list[i] = min;
        }
        return list;
    }

    public static void main(String[] args) {
        int[] myList = {10,34,2,56,7,67,88,42};
        int[] sort = selectionSort(myList);
        for(int i : sort) {
            System.out.print(i);
            System.out.print(", ");
        }
    }
}
